System/Console/Haskeline/InputT.hs

```diff
@@ -65,6 +65,24 @@ instance ( Fail.MonadFail m ) => Fail.MonadFail (InputT m) where
 instance ( MonadFix m ) => MonadFix (InputT m) where
     mfix f = InputT (mfix (unInputT . f))
 
+-- | Run an action in the underlying monad with a function for restoring the
+-- current 'InputT' context.
```

Comment:
Could we elaborate the comment a little more, indicating that it's intended for use with other monad transformer stacks? Maybe even call out the unliftio package explicitly as an example?